### ISSUE TITLE
Exception compiling iterator of tuples with Enum

### ISSUE THREAD

**Issue by vdisasm** [dotnet/roslyn#20164]
**Version Used**:  [Example](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABADAAmwEYAWAbgFgAoauAOwFcx8BRRsI6gb2oEgIk+YIIDG1AL7Vq2AEz4Awtz7YAzISIqAPAAo2TIvgEBLOjCEBKAHytTCI3CjbzfHlV69iB7AHZ823exE5kSCMuaUbpJUUUA) 19 May 2017

Probably related to: https://github.com/dotnet/roslyn/issues/14888

**Steps to Reproduce**:

```
using System.Collections.Generic;

enum Enum1
{
	a, b, c
}

class C
{
	public IEnumerable<(Enum1 a,int b)> Entries()
	{
		yield return ((Enum1)1, 2);
	}
}
```

**Expected Behavior**:
Compiles fine.

**Actual Behavior**:
warning AD0001: 
Analyzer 'Microsoft.CodeAnalysis.CSharp.Diagnostics.RemoveUnnecessaryCast.CSharpRemoveUnnecessaryCastDiagnosticAnalyzer' 

threw an exception of type 'System.InvalidCastException' 

with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.BoundTupleLiteral' 

to type 'Microsoft.CodeAnalysis.CSharp.BoundConvertedTupleLiteral'.'.

**Comment by jcouv** [comment]
I will go ahead and close as dupicate of https://github.com/dotnet/roslyn/issues/17841
The issue is fixed in Visual Studio 2017 version 15.3.
Feel free to re-open with details about the version of Visual Studio 2017 that you are using if the issue still repros in 15.3.

**Comment by jcouv** [comment]
Thanks for reporting the issue.

Here are the [previous reports](https://github.com/dotnet/roslyn/issues?q=is%3Aissue+CSharpRemoveUnnecessaryCastDiagnosticAnalyzer+is%3Aclosed) of a similar crash, in case you want to see more details about the fix.
